sklearn/utils/validation.py

```diff
@@ -24,6 +24,8 @@
 from ..externals.joblib import Memory
 
 
+integer_types = (numbers.Integral, np.integer)
+floating_types = (float, np.floating)
```

Comment:
Those two constants should be moved to `sklearn.utils.fixes` and renamed to:

```python
# In Python 1.8.2 np.integer is not yet a subclass of numbers.Integral
SCALAR_INTEGRAL_TYPES = (numbers.Integral, np.integer)
SCALAR_REAL_TYPES = (numbers.Real, np.floating)
```

Reply:
Ideally, it would be nice to find out the version of numpy that made it such that is scalar types are subclasses of the corresponding Python scalar base types from `numbers`.

Reply:
> # In Python 1.8.2 np.integer is not yet a subclass of numbers.Integral

I think that you mean Numpy, I could not find Python 1.* in conda :)

Reply:
>  it would be nice to find out the version of numpy that made it such that is scalar types are subclasses of the corresponding Python scalar

it is 1.9

Reply:
@ogrisel I disagree with the scalar_real_types. Since integers derive from real whereas they do not derive from float.
```py
>>> import numbers
>>> isinstance(int(42), numbers.Real)
True
>>> isinstance(int(42), float)
False
```